tests/test_directives.py

```
@@ -123,45 +124,90 @@ class TestDirectives:
         with pytest.raises(TypeError):
             directives._add_directive('Test')
         with pytest.raises(ValueError):
-            directives._add_directive(NP)
+            directives._add_directive(PROCESS_FRACTION)
 
-    def test_set_directives(self, setUp, available_directives_list):
-        directives = setUp(available_directives_list=available_directives_list[:-1])
-        directives[OMP_NUM_THREADS.name] = OMP_NUM_THREADS.default
-        assert directives[OMP_NUM_THREADS.name] == OMP_NUM_THREADS.default
+    def test_set_defined_directive(self, setUp, available_directives_list):
+        directives = setUp(available_directives_list=available_directives_list)
+        directives._set_defined_directive(NP.name, 10)
+        assert directives[NP.name] == 10
 
-    def test_define_directive(self, setUp, available_directives_list):
-        directives = setUp(available_directives_list=available_directives_list[:-1])
-        # The values below corresponds to NP, NRANKS, NGPU, EXECUTABLE.
-        valid_values = [100, 100, 100, 'My String']
-        invalid_values = [0, -1, -1, [0]]
-        for idx, dir in enumerate(available_directives_list[:-1]):
-            directives._set_defined_directive(dir.name, valid_values[idx])
-            assert directives[dir.name] == valid_values[idx]
-        for idx, dir in enumerate(available_directives_list[:-1]):
-            with pytest.raises(ValueError):
-                directives._set_defined_directive(dir.name, invalid_values[idx])
-
-    def test_del_directive(self, setUp, available_directives_list):
-        directives = setUp(available_directives_list=available_directives_list[:-1])
-        directives[OMP_NUM_THREADS.name] = OMP_NUM_THREADS.default
+    def test_set_defined_directive_invalid(self, setUp, available_directives_list):
+        directives = setUp(available_directives_list=available_directives_list)
+        directives._set_defined_directive(NP.name, 0)
```

I believe this should raise an error.